ocs_ci/utility/kms.py

```diff
@@ -852,4 +899,29 @@ def get_encryption_kmsid():
 
     for key in csi_kms_conf.get().get("data").keys():
         if constants.VAULT_KMS_PROVIDER in key:
-            return key
+            kmsid.append(key)
+    return kmsid
+
+
+def remove_kmsid(kmsid):
+    """
+    This function will remove all the details for the given kmsid from the csi-kms-connection-details configmap
+
+    Args:
+        kmsid (str) : kmsid to be remove_kmsid
+
+    Raises:
+        KMSResourceCleaneupError: If the kmsid entry is not deleted
+
+    """
+    ocp_obj = ocp.OCP()
+    patch = f'\'[{{"op": "remove", "path": "/data/{kmsid}"}}]\''
+    patch_cmd = (
+        f"patch -n {constants.OPENSHIFT_STORAGE_NAMESPACE} cm "
+        f"{constants.VAULT_KMS_CSI_CONNECTION_DETAILS} --type json -p " + patch
+    )
+    ocp_obj.exec_oc_cmd(command=patch_cmd)
+    kmsid_list = get_encryption_kmsid()
+    if any(kmsid in k for k in kmsid_list):
```

Comment:
Python  behavior looks slightly different
```python
l=["2-vault", "11-vault"]
kmsid="1-vault"

In [15]: kmsid in l                                                                                                                     
Out[15]: False

In [19]: if any(kmsid in k for k in l): 
    ...:     print("present") 
    ...:                                                                                                                                
present
```
Could you please check those cases and modify the code

Reply:
so string membership with list VS string membership with another string behaves differently in python
```python
In [36]: "-vault" in k                                                                                                                  
Out[36]: True

In [37]: k                                                                                                                              
Out[37]: '11-vault'

In [38]: l                                                                                                                              
Out[38]: ['2-vault', '11-vault']

In [39]: "-vault" in l                                                                                                                  
Out[39]: False
```

String VS String membership compares just the substring where as String membership in a list compares the individual element of the list .

Reply:
Tested and changed the code accordingly